GoCD 19.1.0 error with plugin 1.4.0-RC11

## Configuration
**GoCD Version** : 19.1.0
**Plugin Version** : 1.4.0-RC11

## Error
Following error occurs :

> Notification update failed for plugin : slack.notifier
> Interaction with plugin with id 'slack.notifier' implementing 'notification' extension failed while requesting for 'stage-status'. Reason: [java/xml/bind/DatatypeConverter]

![error_message](https://user-images.githubusercontent.com/546128/52718811-4c09d200-2f72-11e9-9f8e-fadee1f48b2e.png)

## Context
I just upgraded GoServer from 18.10.0 to 19.1.0 and this error occured.

## Logs

**go-server.log**
![go-server log](https://user-images.githubusercontent.com/546128/52718911-87a49c00-2f72-11e9-9796-76f59a1f5917.png)

**plugin-slack.notifier.log**
![plugin-slack notifier log](https://user-images.githubusercontent.com/546128/52718940-9723e500-2f72-11e9-85bd-feaf8f36ac4e.png)

## Plugin Configuration

I'm using this plugin to push notifications to both Slack and Rocket. This was working fine on GoCD version 18.10.0

```
# Configuration file for gocd-slack-build-notifier
# https://github.com/ashwanthkumar/gocd-slack-build-notifier

gocd.slack {
  login = "..."
  password = "..."

  server-host = "http://...:8153"
  api-server-host = "http://...:8153"

  webhookUrl = "https://hooks.slack.com/services/..."

  # optional fields
  channel = "#gocd"
  display-console-log-links = true
  displayMaterialChanges = true
  process-all-rules = true

  pipelines = [

  # =====
  # SLACK
  # =====

  # ALL except pipeline xxx (because xxx is triggered on every 30 minutes)
  {
    name = "^(?:(?!(?i)xxx).)*$"
    stage = ".*"
    state = "all" # acceptable values are passed, failed, cancelled, building, fixed, broken or all
    webhookUrl = "https://hooks.slack.com/services/..."
  },

  #  FAILED xxx
  {
    name = "xxx"
    stage = ".*"
    state = "failed"
    webhookUrl = "https://hooks.slack.com/services/..."
  },

  # ===========
  # ROCKET.CHAT
  # ===========

  # ALL except pipeline xxx (because xxx is triggered on every 30 minutes)
  {
    name = "^(?:(?!(?i)xxx).)*$"
    stage = ".*"
    state = "all" # acceptable values are passed, failed, cancelled, building, fixed, broken or all
    webhookUrl = "https://rocket.../hooks/..."
  },

  #  FAILED xxx
  {
    name = "xxx"
    stage = ".*"
    state = "failed"
    webhookUrl = "https://rocket.../hooks..."
  }

  ]
}
````

@arvindsv I installed the RC12 version on my GOCD server (**Windows x64**) instance but with errors. I tried on a clean GOCD server (**Windows x64**) and still getting errors: **zip END header not found**


Here's the **go-server.log** I got :

```console
2019-03-09 14:44:19,377 WARN  [Thread-79] GoPluginDescriptorBuilder:69 - Could not load plugin with jar filename:gocd-slack-notifier-1.4.0-RC12.jar
java.util.zip.ZipException: zip END header not found
	at java.base/java.util.zip.ZipFile$Source.zerror(ZipFile.java:1529)
	at java.base/java.util.zip.ZipFile$Source.findEND(ZipFile.java:1430)
	at java.base/java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1437)
	at java.base/java.util.zip.ZipFile$Source.<init>(ZipFile.java:1268)
	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1231)
	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:726)
	at java.base/java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:843)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:246)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:176)
	at java.base/java.util.jar.JarFile.<init>(JarFile.java:346)
	at java.base/java.util.jar.JarFile.<init>(JarFile.java:317)
	at java.base/java.util.jar.JarFile.<init>(JarFile.java:283)
	at com.thoughtworks.go.plugin.infra.plugininfo.GoPluginDescriptorBuilder.build(GoPluginDescriptorBuilder.java:58)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.pluginJarAdded(DefaultPluginJarChangeListener.java:61)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.lambda$pluginJarAdded$0(DefaultPluginJarLocationMonitor.java:283)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.doOnAllPluginJarChangeListener(DefaultPluginJarLocationMonitor.java:304)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.pluginJarAdded(DefaultPluginJarLocationMonitor.java:283)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.notifyListenersOfAddedPlugins(DefaultPluginJarLocationMonitor.java:217)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.loadAndNotifyPluginsFrom(DefaultPluginJarLocationMonitor.java:208)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.oneShot(DefaultPluginJarLocationMonitor.java:196)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.run(DefaultPluginJarLocationMonitor.java:181)
2019-03-09 14:44:20,704 ERROR [Thread-79] ZipUtil:160 - Failed to unzip file [lib/jaxb-api-2.3.0.jar] to directory [c:\goserver\plugins_work\gocd-slack-notifier-1.4.0-RC12.jar]
java.io.EOFException: Unexpected end of ZLIB input stream
	at java.base/java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:245)
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:159)
	at java.base/java.util.zip.ZipInputStream.read(ZipInputStream.java:195)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2314)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:2270)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2291)
	at com.thoughtworks.go.util.ZipUtil.extractTo(ZipUtil.java:142)
	at com.thoughtworks.go.util.ZipUtil.unzip(ZipUtil.java:120)
	at com.thoughtworks.go.util.ZipUtil.unzip(ZipUtil.java:127)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.explodePluginJarToBundleDir(DefaultPluginJarChangeListener.java:144)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.addPlugin(DefaultPluginJarChangeListener.java:96)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.pluginJarAdded(DefaultPluginJarChangeListener.java:65)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.lambda$pluginJarAdded$0(DefaultPluginJarLocationMonitor.java:283)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.doOnAllPluginJarChangeListener(DefaultPluginJarLocationMonitor.java:304)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.pluginJarAdded(DefaultPluginJarLocationMonitor.java:283)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.notifyListenersOfAddedPlugins(DefaultPluginJarLocationMonitor.java:217)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.loadAndNotifyPluginsFrom(DefaultPluginJarLocationMonitor.java:208)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.oneShot(DefaultPluginJarLocationMonitor.java:196)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.run(DefaultPluginJarLocationMonitor.java:181)
2019-03-09 14:44:20,705 WARN  [Thread-79] DefaultPluginJarLocationMonitor:306 - Plugin listener failed
java.lang.RuntimeException: Failed to copy plugin jar plugins\external\gocd-slack-notifier-1.4.0-RC12.jar to bundle location plugins_work\gocd-slack-notifier-1.4.0-RC12.jar
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.explodePluginJarToBundleDir(DefaultPluginJarChangeListener.java:146)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.addPlugin(DefaultPluginJarChangeListener.java:96)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.pluginJarAdded(DefaultPluginJarChangeListener.java:65)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.lambda$pluginJarAdded$0(DefaultPluginJarLocationMonitor.java:283)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.doOnAllPluginJarChangeListener(DefaultPluginJarLocationMonitor.java:304)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread$DoOnAllListeners.pluginJarAdded(DefaultPluginJarLocationMonitor.java:283)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.notifyListenersOfAddedPlugins(DefaultPluginJarLocationMonitor.java:217)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.loadAndNotifyPluginsFrom(DefaultPluginJarLocationMonitor.java:208)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.oneShot(DefaultPluginJarLocationMonitor.java:196)
	at com.thoughtworks.go.plugin.infra.monitor.DefaultPluginJarLocationMonitor$PluginLocationMonitorThread.run(DefaultPluginJarLocationMonitor.java:181)
Caused by: java.io.EOFException: Unexpected end of ZLIB input stream
	at java.base/java.util.zip.InflaterInputStream.fill(InflaterInputStream.java:245)
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:159)
	at java.base/java.util.zip.ZipInputStream.read(ZipInputStream.java:195)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2314)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:2270)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2291)
	at com.thoughtworks.go.util.ZipUtil.extractTo(ZipUtil.java:142)
	at com.thoughtworks.go.util.ZipUtil.unzip(ZipUtil.java:120)
	at com.thoughtworks.go.util.ZipUtil.unzip(ZipUtil.java:127)
	at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.explodePluginJarToBundleDir(DefaultPluginJarChangeListener.java:144)
	... 9 common frames omitted

```

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>in.ashwanthkumar</groupId>
     <artifactId>gocd-slack-notifier</artifactId>
-    <version>1.4.0-RC11</version>
+    <version>1.4.0-RC12</version>
     <packaging>jar</packaging>
 
     <properties>
@@ -44,6 +44,12 @@
             <groupId>com.typesafe</groupId>
             <artifactId>config</artifactId>
             <version>1.2.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
+            <artifactId>jaxb-api</artifactId>
+            <version>2.3.0</version>
         </dependency>
 
 

